rearrange_cars_luiza.py

```diff
@@ -0,0 +1,253 @@
+        '''
+
+        move_str = 'move from {0} to {1}'.format(from_idx, self.empty_spot_position)
+        if self.print_console:
+            print move_str
+        else:
+            logging.info(move_str)
+
+    def create_log_file(self):
+
+        '''In case the number of moves is more than 100
+        writes the output to the logfile instead of the console
+        '''
+
+        logging.warning('Since the number of cars to arrange is {0} and more than 1000\
+                        , the output with movements will be logged to the file\
+                        '.format(len(self.initial_car_order)))
+        logging.info('Initial car order:')
+        logging.info(self.initial_car_order)
+        logging.info('Final car order:')
+        logging.info(self.final_car_order)
+        logging.info('Moves:')
+
+    def move_car_to_empty_slot(self, car_to_move_position):
+
+        '''Arranges the car with car_to_move_position index to the empty slot
+        Arguments:
+            car_to_move_position: the position from where the car is moved
+        Returns:
+            None
+        '''
+
+        car_to_move = self.current_car_order[car_to_move_position]
+        self.current_car_order[self.empty_spot_position] = car_to_move
+        self.current_car_order[car_to_move_position] = 0
+
+        self.car_position_dict[car_to_move] = self.empty_spot_position
+        self.empty_spot_position = car_to_move_position
+
+    def get_not_arranged_car_index(self):
+
+        '''Finds first car index not yet standing on the final destination
+        index should never reach the end since there should be at least
+        one car not yet arranged
+        '''
+
+        index = 0
+        while self.current_car_order[index] == self.final_car_order[index]:
+            index += 1
+            if index == self.empty_spot_position:
+                index += 1
+
+        return index
+
+    def rearrange_one_car(self):
+
+        '''Puts to the current empty space the car
+        that should be on this place according to the final arrangement
+        if current empty space position is different from the final order empty
+        space position
+
+        Otherwise picks up the random not arranged car and places it to the empty space
+        '''
+
+        if self.final_car_order[self.empty_spot_position] == 0:
+            car_to_move_position = self.get_not_arranged_car_index()
+            car_to_move = self.current_car_order[car_to_move_position]
+        else:
+            car_to_move = self.final_car_order[self.empty_spot_position]
+            car_to_move_position = self.car_position_dict[car_to_move]
+            self.min_num_steps_left -= 1
+
+        self.print_one_step(car_to_move_position)
+        self.move_car_to_empty_slot(car_to_move_position)
+
+    def reset_variables_to_init_state(self):
+
+        '''Resets variables to the same state when class was initialized
+        Needed for running rearrange_all_cars function again

```

Comment:
Right, if I call the function more than once, then some variables and states should be set to the default state. 
That is why I made this additional reset_variables_to_init_state function )